cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -215,6 +215,37 @@ function apply_sriov_node_policy {
   return 0
 }
 
+function setns_sriov_interfaces {
+  #move the pfs to the node
+  mkdir -p /var/run/netns/
+  export pid="$(docker inspect -f '{{.State.Pid}}' $SRIOV_NODE)"
+  ln -sf /proc/$pid/ns/net "/var/run/netns/$SRIOV_NODE"
+
+  sriov_pfs=( /sys/class/net/*/device/sriov_numvfs )
+
+  for ifs in "${sriov_pfs[@]}"; do
+    ifs_name="${ifs%%/device/*}"
+    ifs_name="${ifs_name##*/}"
+
+    if [ $(echo "${PF_BLACKLIST[@]}" | grep "${ifs_name}") ]; then
+      continue
+    fi
+
+    # We set the variable below only in the first iteration as we need only one PF
+    # to inject into the Network Configuration manifest. We need to move all pfs to
+    # the node's namespace and for that reason we do not interrupt the loop.
+    if [ -z "$NODE_PF" ]; then
+      # These values are used to populate the network definition policy yaml.
+      # We just use the first suitable pf
+      # We need the num of vfs because if we don't set this value equals to the total, in case of mellanox
+      # the sriov operator will trigger a node reboot to update the firmware
+      export NODE_PF="$ifs_name"
+      export NODE_PF_NUM_VFS=$(cat /sys/class/net/"$NODE_PF"/device/sriov_totalvfs)
+    fi
```

Comment:
This part is a side effect in this new function.
It has nothing to do with the movement of the PF to the netns of the node.

One option is for the function to return the list of interfaces that moved and from there to continue at the caller.

Also, except this variable, the others seem to be used only in this function so why do we need the "exported"? Declaring them local will make it clearer that they are used only here.

Reply:
> This part is a side effect in this new function.
> It has nothing to do with the movement of the PF to the netns of the node.
> 
> One option is for the function to return the list of interfaces that moved and from there to continue at the caller.
> 
> Also, except this variable, the others seem to be used only in this function so why do we need the "exported"? Declaring them local will make it clearer that they are used only here.

Both `NODE_PF NODE_PF` ,`NODE_PF` need to be exported, since they are used with envsubst
https://github.com/kubevirt/kubevirtci/pull/501/files#diff-499cff045d8a875e8b39f4a9fd59354eb99e8f8d91a95ebc70fbf42d2d5b5beaL209
Will change the others to be local.

Reply:
Now that this is a function, I hope you see why this is problematic.
Please try to leave this function with no side-effects. As this code needs to be taken out now, I hope it will sort everything out.

Reply:
>This part is a side effect in this new function.
It has nothing to do with the movement of the PF to the netns of the node.

We will need deeper refactor, because once we moved the pf, in case we need to get NODE_PF_NUM_VFS,
we will need to find it via `/sys/devices/pci` instead of `/sys/class/net/`.
I will try to do that in case you think it deserves it?

> Also, except this variable, the others seem to be used only in this function so why do we need the "exported"? Declaring them local will make it clearer that they are used only here.

Both `NODE_PF NODE_PF`, `NODE_PF` need to be exported, since they are used with envsubst
```
  # Substitute $NODE_PF and $NODE_PF_NUM_VFS and create SriovNetworkNodePolicy CR
  local -r policy=$(envsubst < $policy_file)
```
Will change the others to be local.

Reply:
> We will need deeper refactor, because once we moved the pf, we will able to get NODE_PF_NUM_VFS,
> only via `/sys/devices/pci` instead of `/sys/class/net/`.
> (unless we peek in the container)
> I will try to do that in case you think it deserves it.

I thought the this suggestion should handle it:
>> One option is for the function to return the list of interfaces that moved and from there to continue at the caller.

> > Also, except this variable, the others seem to be used only in this function so why do we need the "exported"? Declaring them local will make it clearer that they are used only here.
> 
> Both `NODE_PF NODE_PF`, `NODE_PF` need to be exported, since they are used with envsubst
> 
> ```
>   # Substitute $NODE_PF and $NODE_PF_NUM_VFS and create SriovNetworkNodePolicy CR
>   local -r policy=$(envsubst < $policy_file)
> ```
> 
> Will change the others to be local.

I think I answered this already. Now that this is a function, doing this from here is a side effect and seems to me like a blocker to extracting this section into its own function.
The prev point hopefully should resolve this. If it does not, a new strategy is needed here.